src/components/MapView/DateSelector/index.tsx

```diff
@@ -43,38 +51,148 @@ function DateSelector({ availableDates = [], classes }: DateSelectorProps) {
   const dispatch = useDispatch();
   const { startDate: stateStartDate } = useSelector(dateRangeSelector);
 
-  const stateStartDateYear = moment(stateStartDate).year();
   const [selectedDate, setSelectedDate] = useState(moment(stateStartDate));
-  const selectedMonth = selectedDate.month();
+  const [dateRange, setDateRange] = useState([
+    {
+      value: 0,
+      label: '',
+      month: '',
+      isFirstday: false,
+    },
+  ]);
+
+  const [timelinePosition, setTimelinePosition] = useState({ x: 0, y: 0 });
+  const [pointerPosition, setPointerPosition] = useState({ x: 0, y: 0 });
+
+  const dateRef = useRef(availableDates);
 
-  const selectedYear = selectedDate.year();
+  const timeLine = useRef(null);
+  const timeLineWidth = get(timeLine.current, 'offsetWidth', 0);
 
+  // Move the slider automatically so that the pointer always visible
   useEffect(() => {
+    let x = 0;
+    if (pointerPosition.x >= dateRange.length * 10 - timeLineWidth) {
+      // eslint-disable-next-line fp/no-mutation
+      x = timeLineWidth - dateRange.length * 10;
+    } else if (pointerPosition.x > timeLineWidth) {
+      // eslint-disable-next-line fp/no-mutation
+      x = -1 * pointerPosition.x + timeLineWidth / 2;
+    }
+    if (
+      -1 * timelinePosition.x > pointerPosition.x ||
+      -1 * timelinePosition.x + timeLineWidth < pointerPosition.x
+    ) {
+      setTimelinePosition({ x, y: 0 });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pointerPosition]);
+
+  // Create timeline range and set pointer position
+  useEffect(() => {
+    const range = Array.from(
+      moment()
+        .range(
+          moment(stateStartDate).startOf('year'),
+          moment(stateStartDate).endOf('year'),
+        )
+        .by('days'),
+    ).map(e => {
```

Comment:
better name than `e`? e makes me think its an event

Reply:
sorry my bad habit. i changed it to `date`.